### DISCUSSION TITLE
Frontend with auth_token behind Traefik

### DISCUSSION THREAD

**Discussion by llaumgui** [Koenkk/zigbee2mqtt#20839] [General]
Hi,
I use Z2M behind Traefik with a SSO auth. I would like to pass automatically the authtoken auth_token from Traefik to don't have to use it when I use Traefik. Is there a way to add a header to do that ?

**Comment by masonh** [comment]
It doesn't look like Z2M Frontend supports header-based token auth, but it can [take the token in a query parameter](https://github.com/nurikk/zigbee2mqtt-frontend/pull/1590). I don't use Traefik myself, but it seems like [a plugin exists](https://plugins.traefik.io/plugins/628c9f24ffc0cd18356a97bd/query-paramter-modification) which can add that token parameter into the backend requests toward Z2M.

NGINX users can just use a typical [rewrite](https://nginx.org/en/docs/http/ngx_http_rewrite_module.html) like `rewrite ^(.*)$ /$1?token=your-secret-token`, but I don't see a similar feature built into Traefik.
